contracts/asset-proxy/contracts/src/libs/LibAssetData.sol

```diff
@@ -108,64 +115,74 @@ library LibAssetData {
     ///     specification.
     /// @return Number of tokens (or token baskets) that the spender is
     ///     authorized to spend.
-    function getAllowance(address owner, address spender, bytes memory assetData)
+    function getAllowance(
+        address owner,
+        address spender,
+        bytes memory assetData
+    )
         public
         view
         returns (uint256 allowance)
     {
-        bytes4 proxyId = LibBytes.readBytes4(assetData, 0);
+        bytes4 proxyId = assetData.readBytes4(0);
 
         if (proxyId == ERC20_PROXY_ID) {
-            address tokenAddress = LibBytes.readAddress(assetData, 16);
-            return IERC20Token(tokenAddress).allowance(owner, spender);
+            address tokenAddress = assetData.readAddress(16);
+            allowance = IERC20Token(tokenAddress).allowance(owner, spender);
         } else if (proxyId == ERC721_PROXY_ID) {
             (, address tokenAddress, uint256 tokenId) = decodeERC721AssetData(assetData);
             IERC721Token token = IERC721Token(tokenAddress);
-            if (spender == token.getApproved(tokenId) || token.isApprovedForAll(owner, spender)) {
-                return 1;
-            } else {
-                return 0;
-            }
+            allowance = (token.getApproved(tokenId) != address(0) || token.isApprovedForAll(owner, spender)) ? 1 : 0;
```

Comment:
Whoops, not sure how that happened.

Reply:
Side note, I think I'm going to make all unlimited allowances return 2^256 - 1, because this reflects the actual max amounts of units that can be transferred in a single order. `getApproved(tokenId) == spender` would just return an allowance of 1 in this case.